### DISCUSSION TITLE
[jetson-nano]: Alternate debug UART port

### DISCUSSION THREAD

**Discussion by elPrac** [OE4T/meta-tegra#874] [Q&A]
Hi everyone!

I want to use the UART2(Tegra-side) instead of the UART1(Tegra-side) as debug port:
![image](https://user-images.githubusercontent.com/17069897/148497958-c47fe57c-5d51-431b-a8ad-fa07591de0ec.png)

The problem is that I don't really know how to do so... my first approach was to change u-boot device tree like so:
```c
/* dev-tree arch/arm/dts/tegra210-p3450-0000.dts */
...
 / {                                                     
     model = "NVIDIA Jetson Nano Developer Kit";
     compatible = "nvidia,p3450-0000", "nvidia,tegra210";
 
     chosen {
         stdout-path = &uartb;
     };

  ...

  // uartb offset tag
 serial@70006040 {
     status = "okay";
 };
```
After boot I can see that cboot keeps printing log messages to UART1 (as expected because I didn't do anything to change this and I don't really know how to do it) but I stop seeing u-boot messages or prompt from this UART1, I was expecting to see u-boot messages from the 40-pin expansion header pins 8&10 but nothing comes up.

I also tried to find a `debug.uart_instance` option in a tegra210*.cfg kind of file, just like @kekiefer mentioned in [this post](https://github.com/OE4T/meta-tegra/discussions/649#discussioncomment-563783) some time ago, but I couldn't find anything in meta-tegra or in my build/tmp/work/.../tegraflash directory nor any of the build/tmp/work/<arch>.

I'm using Jetson-nano B01 dev-kit and dunfell meta-tegra BSP version, I will really appreciate any clue on the subject, thank you so much for your time.

**Comment by madisongh** [comment]
A quick glance at the sources makes it look like it's at least theoretically possible to do this for U-Boot.  Check `include/configs/p3450-0000.h` - it looks like there is a compile-time option for enabling the other UART there.

For the other components (BPMP, etc.), I'm not sure they're configurable on the t210 platforms.

In fact, before you go too far down this path, you might want to ask about it on the NVIDIA developer forum to see if it's even feasible.

**Reply by elPrac** [reply]
Thank you so much for your time @madisongh, I already tried to change p3450-0000.h but this is what i get from Uboot from UART2, I keep looking into drivers/serial/serial-uclass.c  to see what might be the problem.
![image](https://user-images.githubusercontent.com/17069897/148652111-ce911c2c-2f4b-437d-96ff-9824cf50561c.png)
Also, I follow your suggestion to ask this in NVIDIA forum, [here](https://forums.developer.nvidia.com/t/jetson-nano-b01-alternate-debug-uart-port/199966) is the link if anyone is curious.
